pypowsybl/loadflow.py

```diff
@@ -168,6 +172,12 @@ def __init__(self,
             self.countries_to_balance = countries_to_balance
         if connected_component_mode is not None:
             self.connected_component_mode = connected_component_mode
+        if specific_parameters is not None:
```

Comment:
For the user we should expose the parameters as a dict.
The simplest way is probably to provide a property getter and a property setter for this dict, which will use the underlying lists.
This will not allow to modify the parameters in place, for ex. like that:
```python
parameters.provider_parameters['prop'] = ['val'] # not working
```
But defining the whole dict will be possible like in the constructor:
```python
parameters.provider_parameters = { 'prop': 'val'}
```
and getting the parameters will return a dict:
```python
>>> parameters.provider_parameters
{ 'prop': 'val'}
```

The underlying lists should be hidden as protected members (leading `_`).

Reply:
Actually, maybe the best option would be break the link between the C struct and the python class:
instead of having `Parameters` inherit `_pypowsybl.LoadFlowParameters`, it should be an independent class.

When we get parameters from java, we would perform a conversion from the second to the first (in particular transforming 2 lists to dict), and when we send parameters to java, we will do the opposite.

This way we could support a straight dict in the exposed  `Parameters` class.